src/platform/react-hooks/src/hooks/usePresence.ts

```diff
@@ -44,19 +59,23 @@ export function usePresence<T = any>(
 
   const onUnmount = useCallback(() => {
     // if connection is in one of inactive states, leave call will produce exception
-    if (channel.state === 'attached' && !INACTIVE_CONNECTION_STATES.includes(ably.connection.state)) {
+    if (channel.state === 'attached' && !INACTIVE_CONNECTION_STATES.includes(connectionState)) {
       channel.presence.leave();
     }
-  }, [channel, ably.connection.state]);
+  }, [channel, connectionState]);
```

Comment:
Not sure about this dependency on `connectionState`. Seems like there is no way to avoid it.
Also, since this is a cleanup funciton, I think it would "lag behind" on the actual connection state, e.g. when connection state changes, we call `onUnmount` as our cleanup function in `useEffect` below, which has previous connection state stored (has stored `connected` value but the actual connection state is `closing` or something)

Reply:
Agree, removed callbacks in https://github.com/ably/ably-js/compare/2743834c0521378015334b8d6af3de4b47058e03..2f9fadd5f8dc7fbc840e2e56f98b3340c721b808